### PR TITLE
feat(ethexe): Initial step of reverse gas model implementation

### PR DESCRIPTION
**Pull request by breathx**

### CODE REVIEW

File: gsys/src/lib.rs [hunk 1]
```diff
@@ -350,20 +350,37 @@ pub struct GasMultiplier {
     value_per_gas: Value,
 }
 
+// TODO: handle overflows.
```
Comment (breathx) [inline gsys/src/lib.rs:353]: 1) handle todo

File: gsys/src/lib.rs [hunk 2]
```diff
@@ -350,20 +350,37 @@ pub struct GasMultiplier {
     value_per_gas: Value,
 }
```
Comment (breathx) [inline gsys/src/lib.rs:352]: open todo about wasm builder syscalls checking

File: ethexe/runtime/common/src/journal.rs [hunk 3]
```diff
@@ -151,43 +150,34 @@ impl<S: Storage> JournalHandler for Handler<'_, S> {
     fn send_dispatch(
         &mut self,
         message_id: MessageId,
-        dispatch: CoreDispatch,
+        dispatch: Dispatch,
```
Comment (breathx) [inline ethexe/runtime/common/src/journal.rs:153]: revert core dispatch

File: ethexe/runtime/common/src/journal.rs [hunk 4]
```diff
@@ -202,36 +192,24 @@ impl<S: Storage> JournalHandler for Handler<'_, S> {
...
-        log::trace!("{:?} was added to waitlist block {block}", dispatch);
+        log::trace!("Adding {:?} to waitlist (#{block})", dispatch);
 
-        self.update_program(destination, |state, storage| {
+        self.update_program(dispatch.destination(), |state, storage| {
             let (queue_hash, pop_id) = Self::pop_queue_message(&state, storage);
 
-            if pop_id != dispatch.id {
+            if pop_id != dispatch.id() {
+                // TODO (breathx): figure out what's it.
```
Comment (breathx) [inline ethexe/runtime/common/src/journal.rs:209]: remove todo

**Review by breathx — COMMENTED** [review]
reviewed with @grishasobol

File: gsys/src/lib.rs (hunk 2 above)
Comment (breathx) [reply gsys/src/lib.rs:352]: #4138

File: gsys/src/lib.rs (hunk 1 above)
Comment (breathx) [reply gsys/src/lib.rs:353]: #4137

File: core-processor/src/configs.rs [hunk 5]
```diff
@@ -24,7 +24,8 @@ use gear_core::{
     pages::WasmPagesAmount,
 };
 use gear_lazy_pages_common::LazyPagesCosts;
-use gear_wasm_instrument::syscalls::SyscallName;
+
```
Comment (grishasobol) [inline core-processor/src/configs.rs:27]: Why?))

**Review by grishasobol — APPROVED** [review]

File: core-processor/src/configs.rs (hunk 5 above)
Comment (breathx) [reply core-processor/src/configs.rs:27]: configs of core-processor takes this enum as args, so its useful to have public reimport here for ease of usage

**Pull request merged**